README.md

```diff
@@ -1,3 +1,226 @@
+* [monii/monii-nikic-fast-route-psr7-middleware](https://github.com/monii/monii-nikic-fast-route-psr7-middleware)
+* [monii/monii-response-assertion-psr7-middleware](https://github.com/monii/monii-response-assertion-psr7-middleware)
+* [mtymek/blast-base-url](https://github.com/mtymek/blast-base-url)
+* [ocramius/psr7-session](https://github.com/Ocramius/PSR7Session)
+* [oscarotero/psr7-middlewares](https://github.com/oscarotero/psr7-middlewares)
+* [php-middleware/block-robots](https://github.com/php-middleware/block-robots)
+* [php-middleware/http-authentication](https://github.com/php-middleware/http-authentication)
+* [php-middleware/log-http-messages](https://github.com/php-middleware/log-http-messages)
+* [php-middleware/maintenance](https://github.com/php-middleware/maintenance)
+* [php-middleware/phpdebugbar](https://github.com/php-middleware/phpdebugbar)
+* [php-middleware/request-id](https://github.com/php-middleware/request-id)
+* [relay/middleware](https://github.com/relayphp/Relay.Middleware)
+
+The primary downside of this interface is that the while the interface itself is
+a callable, there is currently no way to type hint a closure in a similar way.
+This may be resolved in the future with [functional interfaces][php-functional].
+
+[php-functional]: https://wiki.php.net/rfc/functional-interfaces
+
+### 4.2 Single Pass (Lambda)
+
+The other approach to middleware is much closer to [StackPHP][stackphp] style
+and is defined as:
+
+
+```
+fn(request, next): response
+```
+
+Middleware taking this approach generally has the following commonalities:
+
+* The middleware is defined as a [callable][php-callable] using a [closure][php-closure]
+  or a class with an `__invoke` method.
+* The middleware is passed 2 arguments during invocation:
+  1. A `RequestInterface` implementation.
+  3. A `callable` that receives the request to dispatch next middleware.
+
+
+In this form, middleware has no access to a response until one is generated by
+innermost middleware. Middleware can then modify the response before returning
+back up the stack.
+
+This approach is often referred to as "single pass" or "lambda" in reference to
+only the request being passed to the middleware.
+
+[php-closure]: http://php.net/closure
+
+#### 4.2.1 Projects Using Single Pass
+
+There are fewer examples of this approach within projects using HTTP Messages,
+with a couple of notable exceptions.
+
+[Guzzle middleware](http://docs.guzzlephp.org/en/latest/handlers-and-middleware.html)
+uses a rather unique approach to middleware that is based around generators and
+promises. Ignoring the generator portion, Guzzle middleware has the signature:
+
+```
+function(RequestInterface $request, array $options): ResponseInterface
+```
+
+[Laravel middleware](https://laravel.com/docs/master/middleware) does make use
+of HTTP Messages but supports middleware with the signature:
```

Comment:
Are you trying to indicate it makes use of *PSR-7* HTTP Messages (which is what you appear to mean everywhere else int the document when you use the verbiage "HTTP Messages")? I'm thinking you meant to say "does not" here...